pkg/manager/member/pd_failover.go

```diff
@@ -140,8 +140,8 @@ func (pf *pdFailover) tryToMarkAPeerAsFailure(tc *v1alpha1.TidbCluster) error {
 			return err
 		}
 
-		pf.recorder.Eventf(tc, apiv1.EventTypeWarning, "PDMemberMarkedAsFailure",
-			"%s(%s) marked as a failure member", podName, pdMember.ID)
+		msg := fmt.Sprintf("pd member[%s] is unhealthy", pdMember.ID)
+		pf.recorder.Event(tc, apiv1.EventTypeWarning, failoverEventReason, fmt.Sprintf(failoverEventMsgPattern, "pd", podName, msg))
```

Comment:
There is a common issue here, as we know, even if the events emitted in the failover/upgrading/scaling functions, it's possible that the sync logic is broken and returned with error without doing the failover/upgrading/scaling, in this case, the event will mismatch with the actual cluster status. What do you think? @weekface @cofyc

Reply:
pd failover method can't be invoked if the status sync is broken.

Reply:
The event for failover seems acceptable from literal (but actually, it's possible that the cluster is not doing failover), but for scaling and upgrading, it's possible that it does not match with the actual cluster status.

Reply:
yes, events should be reported when something has actually happens, e.g. 

- observes something happened or state changes in the actual world (e.g. a PD member becomes unhealth)
- has done some operations (e.g. tidb-operator itself performed some modification requests)

in this place, we added this PD member into the failure member list, but `tidbcluster` object has not been updated yet and maybe skipped if the later operations fail (e.g. when syncing tidb/tikv). 

we can report a PD becomes unhealthy here (we have observed this), but should not report the PD has been marked as a failure member or during the PD failover (we haven't done this yet).

Reply:
This sounds more reasonable. updated.